app/models/user.rb

```diff
@@ -13,6 +14,36 @@ class User < ApplicationRecord
   validates :password, presence: true,
                        length: {minimum: Settings.length.min_pass}
 
+  def self.digest string
+    cost = if ActiveModel::SecurePassword.min_cost
+             BCrypt::Engine::MIN_COST
+           else
+             BCrypt::Engine.cost
+           end
+    # cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+    #                                               BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
bỏ () + dùng method update_attributes nhé em

Reply:
[Reply aid=3224387 to=55448983-1072358421862326272] Tran Thi Hang
à, rollback vì cái đó sẽ validate dữ liệu thôi em, em có thể thêm đoạn allow_nil: true vào validate của password trong model User thử nhé (anh đoán thêm vào là hết lỗi rollback), và em có thể tìm hiểu tại sao nhé (lol)